.ci/CI/src/modules/IntegTest/Graph.jl

```diff
@@ -0,0 +1,239 @@
+- `origin::Vector{String}`: List of already visited packages
+
+# Returns
+
+Dict with the dependency graph. A leaf node has an empty dict. Duplications of dependencies are
+possible.
+"""
+function _build_qed_dependency_graph!(
+        repository_base_path::AbstractString,
+        compat_changes::Dict{String, String},
+        custom_urls::Dict{String, String},
+        package_name::AbstractString,
+        origin::Vector{String},
+    )::Dict
+    qed_dependency_graph = Dict()
+    repository_path = joinpath(repository_base_path, package_name)
+    if !isdir(repository_path)
+        if haskey(custom_urls, package_name)
+            _git_clone(custom_urls[package_name], repository_path)
+        else
+            _git_clone(
+                "https://github.com/QEDjl-project/$(package_name).jl",
+                "dev",
+                repository_path,
+            )
+        end
+    end
+
+    # read dependencies from Project.toml and clone next packages until no
+    # QED dependencies are left
+    project_toml = TOML.parsefile(joinpath(repository_path, "Project.toml"))
+
+    # add package version to compat entires, that they can be set correctly in the compat section of
+    # other packages
+    if !(project_toml["name"] in keys(compat_changes))
+        compat_changes[project_toml["name"]] = project_toml["version"]
+    end
+
+    if haskey(project_toml, "deps")
+        for dep_pkg in keys(project_toml["deps"])
+            # check for circular dependency
+            # actual there should be no circular dependency in graph
+            # if there is a circular dependency in the graph, find a good way to appease the CI
+            # developer
+            if dep_pkg in origin
+                dep_chain = ""
+                for dep in origin
+                    dep_chain *= dep * " -> "
+                end
+                throw(
+                    ErrorException(
+                        "detect circular dependency in graph: $(dep_chain)$(dep_pkg)"
+                    ),
+                )
+            end
+            # handle only dependency starting with QED
+            if startswith(dep_pkg, "QED")
+                qed_dependency_graph[dep_pkg] = _build_qed_dependency_graph!(
+                    repository_base_path,
+                    compat_changes,
+                    custom_urls,
+                    dep_pkg,
+                    vcat(origin, [dep_pkg]),
+                )
+            end
+        end
+    end
+
+    return qed_dependency_graph
+end
+
+function get_qed_integration_test_package_names(
+        test_package::TestPackage,
+        custom_urls::Dict{String, String}
+    )::Vector{String}
+    qed_path = mktempdir(; cleanup = false)
+    compat_changes = Dict{String, String}()
+
+    pkg_tree = build_qed_dependency_graph!(qed_path, compat_changes, custom_urls)
+    return IntegrationTests.depending_projects(
+        test_package.name, r"^QED*|^QuantumElectrodynamics$", pkg_tree
+    )
```

Comment:
I think it would be a good idea to have this regex less hard-coded, in case we will have future packages that do not follow this specific pattern.

Reply:
I would make it configurable, if we have a need for it, ether because for production use or for debugging/development. For example, I can set different Julia versions via environment variables only, because I need it for development.

Else I would follow KISS.

Reply:
Okay, that seems fair.